### ISSUE TITLE
Button cant display ellipsis

### ISSUE THREAD

**Issue by MarvelSQ** [rsuite/rsuite#3241]
### What version of rsuite are you using?

5.35

### What version of React are you using?

17.0.2

### What browser are you using?

Chrome

### Describe the Bug

```tsx
<Button
     style={{
     width: 200,
     overflow: "hidden",
     textOverflow: "ellipsis",
     whiteSpace: "nowrap"
  }}
>
  long text expect to be end with ellipsis
</Button>
```

`Flexbox` causes text to overflow without ellipsis display, and `textAlign` also fails.

### To Reproduce

https://codesandbox.io/s/cool-poincare-cvhpw5?file=/src/index.js

**Comment by Nikhilh26** [comment]
Is it still open?

**Comment by SevenOutman** [comment]
> Is it still open?

Yes, because we haven't decided whether this should be considered an real issue. To how I think about a button, it's essentially not a widget suitable for long content (long enough to overflow). Could you remind me of a real-world example that we should use a button but do not need to make sure user could read the complete content of it?
